code/modules/mob/living/simple_animal/hostile/megafauna/swarmer.dm

```diff
@@ -0,0 +1,314 @@
+#define MEDAL_PREFIX "Swarmer Beacon"
+
+/*
+
+Swarmer Beacon
+
+A strange machine appears anywhere a normal lavaland mob can it produces a swarmer at a rate of
+1/15 seconds, until there are GetTotalAISwarmerCap()/2 swarmers, after this it is up to the swarmers themselves to
+increase their population (it will repopulate them should they fall under GetTotalAISwarmerCap()/2 again)
+
+tl;dr A million of the little hellraisers spawn (controlled by AI) and try to eat mining
+
+Loot: Not much, besides a shit load of artificial bluespace crystals, Oh and mining doesn't get eaten
+that's a plus I suppose.
+
+Difficulty: Special
+
+*/
+
+var/global/list/mob/living/simple_animal/hostile/swarmer/ai/AISwarmers = list()
+var/global/list/mob/living/simple_animal/hostile/swarmer/ai/AISwarmersByType = list()//AISwarmersByType[.../resource] = list(1st, 2nd, nth), AISwarmersByType[../ranged] = list(1st, 2nd, nth) etc.
+var/global/list/AISwarmerCapsByType = list(/mob/living/simple_animal/hostile/swarmer/ai/resource = 30, /mob/living/simple_animal/hostile/swarmer/ai/ranged_combat = 20, /mob/living/simple_animal/hostile/swarmer/ai/melee_combat = 10)
+
+
+//returns a type of AI swarmer that is NOT at max cap
+//type order is shuffled, to prevent bias
+/proc/GetUncappedAISwarmerType()
+	var/static/list/swarmerTypes = subtypesof(/mob/living/simple_animal/hostile/swarmer/ai)
+	LAZYINITLIST(AISwarmersByType)
+	for(var/t in shuffle(swarmerTypes))
+		var/list/amount = AISwarmersByType[t]
+		if(!amount || amount.len <  AISwarmerCapsByType[t])
+			return t
+
+
+//Total of all subtype caps
+/proc/GetTotalAISwarmerCap()
+	var/static/list/swarmerTypes = subtypesof(/mob/living/simple_animal/hostile/swarmer/ai)
+	. = 0
+	LAZYINITLIST(AISwarmersByType)
+	for(var/t in swarmerTypes)
+		. += AISwarmerCapsByType[t]
+
+
+/mob/living/simple_animal/hostile/megafauna/swarmer_swarm_beacon
+	name = "swarmer beacon"
+	desc = "That name is a bit of a mouthful, but stop paying attention to your mouth they're eating everything!"
+	icon = 'icons/mob/swarmer.dmi'
+	icon_state = "swarmer_console"
+	health = 3000
+	maxHealth = 3000 //low-ish HP because it's a passive boss, and the swarm itself is the real foe
+	medal_type = MEDAL_PREFIX
+	score_type = SWARMER_BEACON_SCORE
+	faction = list("mining", "boss", "swarmer")
+	weather_immunities = list("lava","ash")
+	stop_automated_movement = TRUE
+	wander = FALSE
+	anchored = TRUE
+	layer = BELOW_MOB_LAYER
+	AIStatus = AI_OFF
+	var/swarmer_spawn_cooldown = 0
+	var/swarmer_spawn_cooldown_amt = 150 //Deciseconds between the swarmers we spawn
+	var/static/list/swarmer_caps
+
+
+/mob/living/simple_animal/hostile/megafauna/swarmer_swarm_beacon/New()
+	..()
+	swarmer_caps = AISwarmerCapsByType //for admin-edits
```

Comment:
The way you have this done, it can't be edited by admins at all. Well, it can, but all they could do is break it. VV doesn't work with lists like this.